src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -98,8 +98,9 @@ class HNSWIndex : public VecSimIndexAbstract<DistType> {
 #endif
 
 #ifdef BUILD_TESTS
-    friend class HNSWIndexSerializer;
     // Allow the following test to access the index size private member.
+    template <typename, typename>
```

Comment:
```suggestion
    template <DataType, DistType>
```

Reply:
This wont compile with 
error: "declaration of template parameter ‘DataType’ shadows template parameter"
error: "declaration of template parameter ‘DistType’ shadows template parameter"
since it is the same as the class's template type names